.github/workflows/terraform.yml

```
@@ -0,0 +1,167 @@
+    inputs:
+      workspaces:
+        description: "A newline-separated list of globs or dependency glob expressions ('workspace-glob : dependency-glob') representing specific workspaces"
+        required: true
+        type: string
+      global_dependencies:
+        description: "A newline-separated list of globs representing dependencies of each workspace. If any of the dependencies have changed then all workspaces will be returned. Applies only to 'push' and 'pull_request' events."
+        required: false
+        type: string
+      relative_to_path:
+        description: "If provided, results will be relative to the given path"
+        required: false
+        type: string
+    secrets:
+      aws_access_key_id:
+        required: true
+      aws_secret_access_key:
+        required: true
+      ssh_private_key:
+        required: true
+
+jobs:
+  minimize-previous-comments:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Minimize previous comments
+        uses: iStreamPlanet/github-actions/minimize-comments@v1.13.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          by-author: github-actions
+          body-includes: |
+            terraform init
+            terraform fmt
+            terraform validate
+            terraform plan
+
+  determine-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build-workspace-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: build-workspace-matrix
+        uses: iStreamPlanet/github-actions/build-workspace-matrix@v1.13.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          workspaces: ${{ inputs.workspaces }}
+          workflow_dispatch_workspace: ${{ github.event.inputs.workspace }}
+          global_dependencies: ${{ inputs.global_dependencies }}
+          relative_to_path: ${{ inputs.relative_to_path }}
+
+  terraform:
+    needs: [determine-matrix]
+    runs-on: ubuntu-latest
+    if: needs.determine-matrix.outputs.matrix != '{"workspace":[]}'
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.determine-matrix.outputs.matrix) }}
+    concurrency: terraform-${{ matrix.workspace }}
+    # Only declare an environment on events that result in 'terraform apply'
+    environment: ${{ (github.event_name == 'push' || github.event_name == 'workflow_dispatch') && matrix.workspace || null }}
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.aws_access_key_id }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.aws_secret_access_key }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      WORKSPACE: clusters/aws/${{ matrix.workspace }}
+      WORKFLOW_RUN_URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install asdf on the Action Runner
+        uses: asdf-vm/actions/setup@v1.1.0
+        with:
+          asdf_branch: "v0.8.0"
```

```suggestion
          asdf_branch: "v0.10.2"
```
Probably can move to the latest: https://github.com/asdf-vm/asdf/releases